### ISSUE TITLE
[windows] undefined symbol createLinalgVectorizePass

### ISSUE THREAD

**Issue by stellaraccident** [google/iree#4485]
Not sure why this is hitting the windows build specifically.

```
2021-01-12T11:53:26.9905937Z iree_compiler_Conversion_LinalgToLLVM_LinalgToLLVM.lib(Passes.cpp.obj) : error LNK2019: unresolved external symbol "class std::unique_ptr<class mlir::OperationPass<class mlir::ModuleOp>,struct std::default_delete<class mlir::OperationPass<class mlir::ModuleOp> > > __cdecl mlir::iree_compiler::createLinalgVectorizePass(void)" (?createLinalgVectorizePass@iree_compiler@mlir@@YA?AV?$unique_ptr@V?$OperationPass@VModuleOp@mlir@@@mlir@@U?$default_delete@V?$OperationPass@VModuleOp@mlir@@@mlir@@@std@@@std@@XZ) referenced in function "void __cdecl mlir::iree_compiler::buildLLVMTransformPassPipeline(class mlir::OpPassManager &)" (?buildLLVMTransformPassPipeline@iree_compiler@mlir@@YAXAEAVOpPassManager@2@@Z)
2021-01-12T11:53:26.9909787Z iree\tools\iree-translate.exe : fatal error LNK1120: 1 unresolved externals
```

In https://pipelines.actions.githubusercontent.com/SaXEyjchYvnWH9Rq3qeZEPLXtO1Hv1A8b0nd0VWR5igNiKNAhf/_apis/pipelines/1/runs/41802/signedlogcontent/13?urlExpires=2021-01-12T17%3A12%3A55.7470976Z&urlSigningMethod=HMACV1&urlSignature=YPZzt%2FgxI9hGOtpFX8x9%2FST%2Fcvnbfj0EnTQaFsZoxdA%3D

**Comment by ThomasRaoux** [comment]
I can't access the link but looking at the code is because I made a mismatch between declaration and definition for the function.

`std::unique_ptr<OperationPass<ModuleOp>> createLinalgVectorizePass();`
vs
std::unique_ptr<FunctionPass> createLinalgVectorizePass() {

I'll send a fix for that but I'm not sure how to reproduce the problem to verify that it actually fixes it. (and somehow the link above doesn't work for me). Is there a way for me to reproduce?

**Comment by benvanik** [comment]
That'd likely be the problem - I've seen this before in that exact situation (wrongly declared pass return types) - I think because some of these files don't properly include the Passes.h headers and as such miss on the "redeclaration with different return types" errors. I did a scrub awhile ago to add them but we should make sure all of the files declaring passes that are listed in Passes.h files *also* include the Passes.h.

**Comment by stellaraccident** [comment]
Lol, finally a windows build issue that is a real problem in the code... vs a mumble..mumble..backslash..powershell..space issue.

**Comment by benvanik** [comment]
yeah - these kind of issues are great reminders that not having a compiler monoculture is a good thing! :P

**Comment by stellaraccident** [comment]
Presumed fixed by #4487
